eth/transactionManager.go

```diff
@@ -228,8 +225,14 @@ func (tm *TransactionManager) checkTxLoop() {
 			receipt, err = tm.wait(tx)
 		}
 
+		if receipt == nil {
+			txReceipt = types.Receipt{}
```

Comment:
I think there are 2 problems with this code snippet that will cause a call to `CheckTx()` to hang forever because [this line](https://github.com/livepeer/go-livepeer/blob/0b2c51b7373ae719bf29bd3921e8c71064d0f406/eth/client.go#L878) would always evaluate to false:

1. The TxHash field for an empty receipt created on this line will be the zero value
2. The TxHash field for a valid receipt on L231 will be set to the hash of the tx that was just mined and not necessarily the hash of the initial tx that was passed to `CheckTx()` if the tx that was just mined was a replacement tx

Problem 1 could be addressed by setting the TxHash field on the receipt on L229. But, this would not address problem 2.

Problem 1 and 2 could be addressed if the `transactionReceipt` struct contained a `OrigTxHash` field that is set to the hash of the first tx that was submitted and if `CheckTx()` is updated to use this `OrigTxHash` field in [this comparison](https://github.com/livepeer/go-livepeer/blob/0b2c51b7373ae719bf29bd3921e8c71064d0f406/eth/client.go#L878) instead of the TxHash field of the underlying receipt.

Reply:
Good catch, however this is not an issue with this particular snippet and would have already existed before. 

The only reason why it manifests now is because that would actually occur due to this snipper rather than a nil-pointer exception. 

Nontheless, will fix !